### ISSUE TITLE
NoMethodError - undefined method `uuid' for nil:NilClass

### ISSUE THREAD

**Issue by ptigas** [CocoaPods/CocoaPods#5685]
Hi,

I've been trying to update the pods of a project that used to work but not anymore. Unfortunately I can't find the cause action of this.

* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

I was trying to update the dependencies by running `pod install`

## What did you expect to happen?

Install all pod dependencies correctly.

## What happened instead?

I got the error below

```
NoMethodError - undefined method `uuid' for nil:NilClass
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.2.0/lib/xcodeproj/project.rb:559:in `map'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.2.0/lib/xcodeproj/project.rb:559:in `block in host_targets_for_embedded_target'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.2.0/lib/xcodeproj/project.rb:557:in `select'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.2.0/lib/xcodeproj/project.rb:557:in `host_targets_for_embedded_target'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.2.0/lib/xcodeproj/project.rb:542:in `block in embedded_targets_in_native_target'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.2.0/lib/xcodeproj/project.rb:541:in `select'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.2.0/lib/xcodeproj/project.rb:541:in `embedded_targets_in_native_target'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:243:in `block in copy_embedded_target_pod_targets_to_host'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:242:in `map'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:242:in `copy_embedded_target_pod_targets_to_host'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:299:in `block in generate_targets'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:298:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:298:in `generate_targets'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:78:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer.rb:238:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer.rb:152:in `block in resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/user_interface.rb:64:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer.rb:151:in `resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer.rb:111:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/command/install.rb:37:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-1.0.0/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/command.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

### Stack

```
   CocoaPods : 1.1.0.beta.1
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin15]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.11.4 (15E65)
       Xcode : 7.3.1 (7D1014)
         Git : git version 2.7.4 (Apple Git-66)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ dbac70ddadd285892e14d9dfebaccb091f7ed27c
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment this line to define a global platform for your project
# platform :ios, '8.0'
# Uncomment this line if you're using Swift
use_frameworks!

project 'CosmoAdmin.xcodeproj'

target 'CosmoAdmin' do
    pod 'Cartography', '~> 0.6'
    pod 'iOSDFULibrary'
end
```

**Comment by segiddins** [comment]
I believe this might be fixed by

**Comment by benasher44** [comment]
Yep I ran into this issue when working on #5637. Should be fixed in that PR.

**Comment by evautherin** [comment]
I'm still running into this issue…

### Command

```
/usr/local/bin/pod install
```

### Report

* What did you do?

* What did you expect to happen?

* What happened instead?


### Stack

```
   CocoaPods : 1.1.0.beta.1
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin15]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.11.6 (15G31)
       Xcode : 7.3.1 (7D1014)
         Git : git version 2.7.4 (Apple Git-66)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 1108c5668800e12bef16ddcf440129808ff42200
```

### Plugins

```
cocoapods-deintegrate     : 1.0.0
cocoapods-plugins         : 1.0.0
cocoapods-search          : 1.0.0
cocoapods-stats           : 1.0.0
cocoapods-trunk           : 1.0.0
cocoapods-try             : 1.1.0
cocoapods-try-release-fix : 0.1.1
```

### Podfile

```ruby
source 'https://github.com/CocoaPods/Specs.git'
use_frameworks!
platform :ios, '10.0'

workspace 'Adanam'

def common_pods
    #pod 'XCGLogger', :git => "https://github.com/alexiosdev/XCGLogger.git", :branch => 'swift3_alex'
    
	pod 'XCGLogger',
		:git => "https://github.com/evautherin/XCGLogger.git",
		:branch => 'patch-1'
		
    pod 'PromiseKit/CorePromise',
    	:git => 'https://github.com/mxcl/PromiseKit.git',
    	:branch => 'swift-3.0'
    # pod 'PromiseKit/Foundation'
    
    pod 'Alamofire',
    	:git => 'https://github.com/Alamofire/Alamofire.git',
    	:branch => 'swift3'
    	
    pod 'ObjectMapper',
    	:git => 'https://github.com/Hearst-DD/ObjectMapper.git',
    	:branch => 'swift-3'
    	
    pod 'AlamofireObjectMapper',
    	:git => 'https://github.com/evautherin/AlamofireObjectMapper.git',
    	:branch => 'swift-3'
    	
    # pod 'SwiftyJSON', :git => 'https://github.com/SwiftyJSON/SwiftyJSON.git', :branch => 'swift3'
    pod 'SwiftKeychainWrapper', :git => 'https://github.com/jrendel/SwiftKeychainWrapper.git', :branch => 'develop'
   	# pod 'Former'
    # pod 'CryptoSwift', :git => 'https://github.com/krzyzanowskim/CryptoSwift.git', :branch => 'swift3'
    
    pod 'M13Checkbox', '~> 1.2'
end


target 'AdanamEngine' do
	project 'AdanamEngine/AdanamEngine.xcodeproj'
    common_pods
end


target 'CloudsBooster' do
	project 'CloudsBooster/CloudsBooster.xcodeproj'
    common_pods

    target 'BoosterKit' do
        inherit! :search_paths
        common_pods
    end

    target 'SaveToCloudsBooster' do
        inherit! :search_paths
        common_pods
    end
end


target 'BlockShare' do
	project 'BlockShare/BlockShare.xcodeproj'
    common_pods

    target 'BlockKit' do
        inherit! :search_paths
        common_pods
    end

    target 'SaveToBlockShare' do
        inherit! :search_paths
        common_pods
    end
end
```

### Error

```
NoMethodError - undefined method `uuid' for nil:NilClass
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.2.0/lib/xcodeproj/project.rb:559:in `map'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.2.0/lib/xcodeproj/project.rb:559:in `block in host_targets_for_embedded_target'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.2.0/lib/xcodeproj/project.rb:557:in `select'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.2.0/lib/xcodeproj/project.rb:557:in `host_targets_for_embedded_target'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:273:in `block (2 levels) in verify_host_targets_in_podfile'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:272:in `map'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:272:in `block in verify_host_targets_in_podfile'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:271:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:271:in `verify_host_targets_in_podfile'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:297:in `generate_targets'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/analyzer.rb:78:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer.rb:238:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer.rb:152:in `block in resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/user_interface.rb:64:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer.rb:151:in `resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer.rb:111:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/command/install.rb:37:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-1.0.0/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/command.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

**Comment by benasher44** [comment]
@evautherin this is a bug in 1.1.0.beta.1, and we haven't done another release yet. You'll have to use master if you want the fix, or roll back to 1.0.1 until another release is put out.

**Comment by evautherin** [comment]
Thank you for your reply
